test/test_torch.py

```diff
@@ -9841,6 +9841,25 @@ def test_add(self, device):
                                r"For integral input tensors, argument alpha must not be a floating point number\.",
                                lambda: torch.add(m1, m2, alpha=1.0))
 
+        # Scalar
+        for dtype in torch.testing.get_all_dtypes():
+            m1 = torch.tensor([1, 2], dtype=dtype, device=device)
+            if dtype == torch.bool:
```

Comment:
can you make the these use the generic test framework at the bottom so we can ensure XLA has these semantics?

Reply:
Are you referring to `test_*(self, device, dtype)` style test cases?

Reply:
yes, that's what I mean.